src/make_round_config.R

```diff
@@ -1 +1,214 @@
+#' and validates it, returning an error if the config is invalid.
+write_and_validate_task_config <- function(task_config, round_id) {
+  hub_dir <- here::here()
+  hubAdmin::write_config(task_config, hub_path = hub_dir, overwrite = TRUE, silent = TRUE)
+  valid_task_config <- hubAdmin::validate_config(
+    hub_path = hub_dir,
+    config = c("tasks"),
+    schema_version = "from_config",
+    branch = "main"
+  )
+  if (isFALSE(valid_task_config)) {
+    print(valid_task_config)
+    stop("Generated task config is invalid for round: ", round_id)
+  }
+}
+
+#' Coerce a list to a round object
+#'
+#' @description
+#' `coerce_to_round` converts a list of round information to a round object,
+#'   using attributes of the newly-created round as needed.
+#'
+#' @param existing_round List. A single hub round, in list form.
+#' @param new_round A Hubverse round created via hubAdmin::create_new_round().
+#' @returns A Hubverse round.
+coerce_to_round <- function(existing_round, new_round) {
+  .Deprecated("hubAdmin::append_round()")
+  class(existing_round) <- c("round", "list")
+  attr(existing_round, "round_id") <- attr(new_round, "round_id")
+  attr(existing_round, "schema_id") <- attr(new_round, "schema_id")
+  existing_round$model_tasks[[1]]$task_ids[["nowcast_date"]][["required"]] <-
+    list(existing_round$model_tasks[[1]]$task_ids[["nowcast_date"]][["required"]])
+  return(existing_round)
+}
+
+#' Create a new task_config that includes existing rounds and a new round
+#'
+#' @description
+#' `append_to_round` coerces the hub's existing rounds to a list of rounds,
+#'   appends the new round to the list, and uses the result to create a new
+#'   task config.
+#'
+#' @param old_task_config List. A hub task config in list form, as created by
+#'   hubAdmin::read_config().
+#' @param new_round A Hubverse round created via hubAdmin::create_new_round().
+#' @returns A Hubverse task config that represents the hub's existing rounds
+#'   plus a newly-created round.
+append_round <- function(old_task_config, new_round) {
+  .Deprecated("hubAdmin::append_round()")
+  existing_round_list <- lapply(old_task_config$rounds, coerce_to_round, new_round)
+  round_list <- append(existing_round_list, list(new_round), 0)
+  rounds <- do.call(hubAdmin::create_rounds, round_list)
+  new_task_config <- hubAdmin::create_config(rounds)
+  return(new_task_config)
+}
+
+new_round <- create_new_round()
+
+existing_task_config <- try(hubUtils::read_config(dirname(getwd()), config = c("tasks")), silent = TRUE)
+if (inherits(existing_task_config, "try-error")) {
+  print("Existing config not found, creating a new tasks.json")
+  new_task_config <- hubAdmin::create_config(hubAdmin::create_rounds(new_round))
+} else {
+  print("Existing config found, adding a new round")
+  new_task_config <- append_round(existing_task_config, new_round)
+}
+
+write_and_validate_task_config(new_task_config, this_round_date)
+print("New round added to tasks.json")
```

Comment:
Minor: use the {cli} package for messages to the user (sends messages to stderr and are easier to visually parse from the output)

```suggestion
  cli::cli_alert_info("Existing config not found, creating a new tasks.json")
  new_task_config <- hubAdmin::create_config(hubAdmin::create_rounds(new_round))
} else {
  cli::cli_alert_info("Existing config found, adding a new round")
  new_task_config <- append_round(existing_task_config, new_round)
}

write_and_validate_task_config(new_task_config, this_round_date)
cli::cli_alert_info("New round added to tasks.json")
```

Reply:
Just pushed a separate commit for this update. It's much nicer--thanks for the tip!